apis/ec2/v1beta1/natgateway_types.go

```diff
@@ -0,0 +1,125 @@
+
+import (
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Defines the states of NatGateway
+const (
+	NatGatewayStatusPending   = "pending"
+	NatGatewayStatusFailed    = "failed"
+	NatGatewayStatusAvailable = "available"
+	NatGatewayStatusDeleting  = "deleting"
+	NatGatewayStatusDeleted   = "deleted"
+)
+
+// NATGatewayParameters defined the desired state of an AWS VPC NAT Gateway
+type NATGatewayParameters struct {
+
+	// TODO(muvaf): Region is a required field but in order to keep backward compatibility
+	// with old Provider type and not bear the cost of bumping to v1beta2, we're
+	// keeping it optional for now. Reconsider before v1beta2 or v1.
+
+	// Region is the region you'd like your NatGateway to be created in.
+	// +immutable
+	Region *string `json:"region,omitempty"`
+
+	// AllocationID is the Elastic IP allocation ID
+	// +immutable
+	// +optional
+	AllocationID *string `json:"allocationId,omitempty"`
+
+	// AllocationIDRef references an EIP and retrieves it's allocation id
+	// +immutable
+	// +optional
+	AllocationIDRef *runtimev1alpha1.Reference `json:"allocationIdRef,omitempty"`
+
+	// AllocationIDSelector references an EIP by selector and retrieves it's allocation id
+	// +immutable
+	// +optional
+	AllocationIDSelector *runtimev1alpha1.Selector `json:"allocationIdSelector,omitempty"`
+
+	// SubnetID is the subnet the NAT gateways needs to be associated to
+	// +immutable
+	// +optional
+	SubnetID *string `json:"subnetId,omitempty"`
+
+	// SubnetIDRef references a subnet and retrives it's subnet id
+	// +immutable
+	// +optional
+	SubnetIDRef *runtimev1alpha1.Reference `json:"subnetIdRef,omitempty"`
+
+	// SubnetIDSelector references a subnet by selector and retrives it's subnet id
+	// +immutable
+	// +optional
+	SubnetIDSelector *runtimev1alpha1.Selector `json:"subnetIdSelector,omitempty"`
+
+	// Tags represents to current ec2 tags.
+	// +optional
+	Tags []Tag `json:"tags,omitempty"`
+}
+
+// NATGatewaySpec defines the desired state of a NAT Gateway
+type NATGatewaySpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  NATGatewayParameters `json:"forProvider"`
+}
+
+// NATGatewayObservation keeps the state for the CR
+type NATGatewayObservation struct {
+	CreateTime          *metav1.Time        `json:"createTime,omitempty"`
+	DeleteTime          *metav1.Time        `json:"deleteTime,omitempty"`
+	FailureCode         string              `json:"failureCode,omitempty"`
+	FailureMessage      string              `json:"failureMessage,omitempty"`
+	NatGatewayAddresses []NATGatewayAddress `json:"natGatewayAddresses,omitempty"`
+	NatGatewayID        string              `json:"natGatewayId,omitempty"`
+	State               string              `json:"state,omitempty"`
+	SubnetID            string              `json:"subnetId,omitempty"`
+	Tags                []Tag               `json:"tags,omitempty"`
```

Comment:
```suggestion
```
We try not to replicate the same fields in both status and spec, see [API Patterns](https://github.com/crossplane/crossplane/blob/master/design/one-pager-managed-resource-api-design.md#high-fidelity). So, `Tags` and `SubnetID` should appear only under spec.